conans/client/cache/cache.py

```diff
@@ -58,18 +58,16 @@ def __init__(self, base_folder, store_folder, output):
         super(ClientCache, self).__init__(self._store_folder)
         self.editable_packages = EditablePackages(self.conan_folder)
 
-    def package_layout(self, ref, short_paths=False):
+    def package_layout(self, ref, short_paths=False, *args, **kwargs):
```

Comment:
If the arguments are not used, please remove them.

Reply:
If I remove them I get a warning saying that this function doesn't match the signature of the one defined in the base class. If I add the missing argument `no_lock=False` I'm telling the caller that it could be set but I won't use it inside the function (I'm not sure if we want to allow different locking behavior for each package, maybe yes). Using `*args, **kwargs` is the only way to avoid the warning without cheating the caller.

If we are to split SimplePath/ClientCache for Server/Client, maybe this is the PR to do it too.

Reply:
Ok, leave it, we could change it later when SimplePaths is decoupled.

No, I think that deserves another PR. This current PR is good as-is, better to keep it focused, not add more refactors to it.